src/query/mlt/mlt.rs

```diff
@@ -0,0 +1,387 @@
+                        .f64_value()
+                        .ok_or(TantivyError::InvalidArgument("invalid value".to_string()))?;
+                    if !self.is_noise_word(val.to_string()) {
+                        let term = Term::from_field_f64(field, val);
+                        *term_frequencies.entry(term).or_insert(0) += 1;
+                    }
+                }
+            }
+            _ => {}
+        }
+        Ok(())
+    }
+
+    /// Determines if the term is likely to be of interest based on "more-like-this" settings
+    fn is_noise_word(&self, word: String) -> bool {
+        let word_length = word.len();
+        if word_length == 0 {
+            return true;
+        }
+        if self
+            .min_word_length
+            .map(|min| word_length < min)
+            .unwrap_or(false)
+        {
+            return true;
+        }
+        if self
+            .max_word_length
+            .map(|max| word_length > max)
+            .unwrap_or(false)
+        {
+            return true;
+        }
+        return self.stop_words.contains(&word);
+    }
+
+    /// Couputes the score for each term while ignoring not useful terms
+    fn create_score_term(
+        &self,
+        searcher: &Searcher,
+        per_field_term_frequencies: HashMap<Term, usize>,
+    ) -> Result<Vec<ScoreTerm>> {
+        let mut score_terms = BinaryHeap::new();
+        let num_docs = searcher
+            .segment_readers()
+            .iter()
+            .map(|x| x.num_docs() as u64)
+            .sum::<u64>();
+
+        for (term, term_frequency) in per_field_term_frequencies.into_iter() {
+            // ignore terms with less than min_term_frequency
+            if self
+                .min_term_frequency
+                .map(|x| term_frequency < x)
+                .unwrap_or(false)
+            {
+                continue;
+            }
+
+            let doc_freq = searcher.doc_freq(&term)?;
+
+            // ignore terms with less than min_doc_frequency
+            if self
+                .min_doc_frequency
+                .map(|x| doc_freq < x)
+                .unwrap_or(false)
+            {
+                continue;
+            }
+
+            // ignore terms with more than max_doc_frequency
+            if self
+                .max_doc_frequency
+                .map(|x| doc_freq > x)
+                .unwrap_or(false)
+            {
+                continue;
+            }
+
+            // ignore terms with zero frequency
+            if doc_freq == 0 {
+                continue;
+            }
+
+            // compute similarity & score
+            let idf = self.idf(doc_freq, num_docs);
+            let score = (term_frequency as f32) * idf;
+            score_terms.push(ScoreTerm::new(term, score));
+        }
+
+        // limit ourself to max_query terms. we need to sort so to avoid discarding important terms
+        let score_terms = if let Some(max_query_terms) = self.max_query_terms {
```

Comment:
needlessly convoluted...

It is ok to use truncate and mutate stuff if it makes your code more readable.

// score_terms assumed sorted
if let Some(max_query_terms) = self.max_query_terms_opt {
    score_terms.truncate(max_query_terms);
}